cpu/stm32l1/lpm_arch.c

```diff
@@ -14,32 +15,92 @@
  * @brief       Implementation of the kernel's lpm interface
  *
  * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Laksh Bhatia <bhatialaksh3@gmail.com>
  *
  * @}
  */
 
+#include "cpu.h"
 #include "arch/lpm_arch.h"
 
+static enum lpm_mode current_mode = LPM_UNKNOWN;
+
 void lpm_arch_init(void)
 {
-    /* TODO */
+    current_mode = LPM_ON;
 }
 
 enum lpm_mode lpm_arch_set(enum lpm_mode target)
 {
-    /* TODO */
-    return 0;
+    enum lpm_mode last_mode = current_mode;
+    switch (target) {
+        case LPM_ON:                    /* STM Run mode */
+            current_mode = LPM_ON;
+            break;
+        case LPM_IDLE:                  /* STM Low power run mode */
+            current_mode = LPM_IDLE;
+            PWR->CR |= (PWR_CR_LPSDSR | PWR_CR_LPRUN);
```

Comment:
Low power run mode cannot be entered if the system clock speed exceeds `fMSI 1` I guess this is 4.194 MHz but all boards are currently using the HSI as clock source and running at >= 32MHz